### ISSUE TITLE
No rich presence display 1.2.0

### ISSUE THREAD

**Issue by Radiicall** [Radiicall/jellyfin-rpc#127]
I'm trying on Windows with the new release and this is the only output with no Rich Presence display?
![image](https://github.com/user-attachments/assets/281d8dd9-3ff1-4b75-a4d1-98726aeb24ea)

Config looks like this:
```json
{
  "jellyfin": {
    "url": "http://xxxxxxxxxxxxxxxxx",
    "api_key": "xxxxxxxxxxxxxxxxxxxxx",
    "username": [
      "XXXXX"
    ],
    "movies": {
      "display": [],
      "separator": "-"
    },
    "music": {
      "display": [],
      "separator": "—"
    },
    "blacklist": null,
    "self_signed_cert": null
  },
  "discord": {
	"show_paused": false,
    "application_id": "XXXXXXXXXXXXXX",
    "buttons": [
            {
                "name": "The Bear's Den",
                "url": "https://thebearsden.live"
            },
            {
                "name": "dynamic",
                "url": "dynamic"
            }
        ]
  },
  "imgur": {
    "client_id": "XXXXXXXXXXXXX"
  },
  "images": {
    "enable_images": true,
    "imgur_images": true
  }
}
```
Put x's in place of sensitive data.
I'm not sure what's going on. The same config works on the previous version fine. Am I missing something here?

_Originally posted by @AkikoKumagara in https://github.com/Radiicall/jellyfin-rpc/issues/120#issuecomment-2241956603_

**Comment by Radiicall** [comment]
@AkikoKumagara can you rerun with --log-level=debug? It should give some more info about where its stopping

**Comment by AkikoKumagara** [comment]
> @AkikoKumagara can you rerun with `--log-level=debug`? It should give some more info about where its stopping

It gets to this and loops the last two lines repeatedly.
This is the correct IP address and port #, and I am playing media via Jellyfin Media Player and have also tried directly through a browser @ that IP with no difference.

2024-07-22 15:27:17 DEBUG [reqwest::connect] starting new connection: http://192.168.1.150:8096/
2024-07-22 15:27:17 DEBUG [hyper_util::client::legacy::connect::http] connecting to 192.168.1.150:8096
2024-07-22 15:27:17 DEBUG [hyper_util::client::legacy::connect::http] connected to 192.168.1.150:8096
2024-07-22 15:27:17 DEBUG [hyper_util::client::legacy::pool] pooling idle connection for ("http", 192.168.1.150:8096)
2024-07-22 15:27:20 DEBUG [hyper_util::client::legacy::pool] reuse idle connection for ("http", 192.168.1.150:8096)

**Comment by Radiicall** [comment]
Are you watching anything on jellyfin when this is happening? Try checking again while having something playing if not

**Comment by AkikoKumagara** [comment]
> Are you watching anything on jellyfin when this is happening? Try checking again while having something playing if not

Yes, it continues unchanged while media plays.

**Comment by Radiicall** [comment]
Does the username in your config file have any capital letters?

**Comment by AkikoKumagara** [comment]
> Does the username in your config file have any capital letters?

It does! Changing it to a lowercase in the config caused this...

`thread 'main' panicked at C:\Users\runneradmin\.cargo\registry\src\index.crates.io-6f17d22bba15001f\jellyfin-rpc-1.2.0\src\lib.rs:242:33:
index out of bounds: the len is 0 but the index is 0
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace`

**Comment by Radiicall** [comment]
ah yes i can see the problem there, can you try playing something else for now, like the movie frozen that you displayed in the previous issue?

**Comment by AkikoKumagara** [comment]
> ah yes i can see the problem there, can you try playing something else for now, like the movie frozen that you displayed in the previous issue?

_Looks like it works now_ 
![image](https://github.com/user-attachments/assets/85078069-15e0-4c9c-a5fe-631ab27558cc)

**Comment by Radiicall** [comment]
Yes, the issue you got the second time around was the buttons, it was trying to access a non-existent dynamic button. Ill push some code changes that will fix all of this. A new update will come out after ive looked at the other issue in the repo

**Comment by AkikoKumagara** [comment]
> Yes, the issue you got the second time around was the buttons, it was trying to access a non-existent dynamic button. Ill push some code changes that will fix all of this. A new update will come out after ive looked at the other issue in the repo

Thanks for the hard work and always being receptive to feedback, I appreciate what you do!

**Comment by Radiicall** [comment]
https://github.com/Radiicall/jellyfin-rpc/actions/runs/10044181135 When all of these are done there should be a windows artifact at the bottom of the page, please try it and see if it resolves the problems you're having

**Comment by AkikoKumagara** [comment]
Working, both with uppercase and without in username now, too.

**Comment by Radiicall** [comment]
I'll keep this open so other people will see it before posting their own, if you see anything else not working then just post that here

**Comment by Radiicall** [comment]
Fixed in 1.2.1
